### DISCUSSION TITLE
ChatSydney To Termux /Original Author By InterestingDarkness

### DISCUSSION THREAD

**Discussion by viopsa233** [viopsa233/AI-testers-Forum#44] [技术交流]
# 准备

Android手机 不需要root
魔法 自行解决 (能来到这里的应该都有了吧......)
一个微软账号

# 开始

安装[Termux](https://github.com/termux/termux-app/releases)
先打开魔法，然后在termux中依次输入（复制粘贴用剪贴板)
```
apt update && apt upgrade
pkg install git(选装,对以后有帮助)
pkg install python-pip
pip install setuptools
```

然后去下载[ChatSydney](https://github.com/InterestingDarkness/ChatSydney)到你的手机根目录(例如:/storage/emulated/0/ChatSydney-zh-cn-master/)

下载完之后记得解压然后把你的cookie放进去,怎么导出[看这个
](https://github.com/viopsa233/AI-testers-Forum/discussions/29#:~:text=%E5%AE%89%E8%A3%85%E2%80%9CCookie%2DEditor,%E4%BF%9D%E5%AD%98%E5%B9%B6%E5%85%B3%E9%97%AD)

然后再次打开Termux,输入`termux-setup-storage`来允许Termux访问手机所有文件
输入`tree`看一下根目录
然后创造文件软链接
`ln -s /storage/emulated/0/xxxxxx/`
(这里是你刚刚解压的ChatSydney目录,例如:ln -s /storage/emulated/0/ChatSydney/)
输入`ls`你会发现多了一个文件夹
![IMG_20230511_215435](https://github.com/viopsa233/AI-testers-Forum/assets/118115208/6c32f4d5-d91d-4dc7-92f4-942d29e5968d)
`cd` 它(例如: cd /storage/emulated/0/ChatSydney/)
然后输入`pip install -r requirements.txt`来安装要求文件
然后输入`python main.py`，会输出一个网址，进入
![IMG_20230511_215855](https://github.com/viopsa233/AI-testers-Forum/assets/118115208/32ab7988-9008-443d-be76-820feeb8cdfa)

开聊!

![Screenshot_2023-05-11-21-58-07-19_a252b927494330c](https://github.com/viopsa233/AI-testers-Forum/assets/118115208/a96e1a4b-70ba-4d31-99dd-7f4fd89cf5ce)
(这是我摸索出来的,刚用Termux,估计还有更简单的方法吧)

**Comment by Viryxs** [comment]
我的超人！🫡

**Reply by 33boku** [reply]
大佬，出现FileNotFoundError: [Errno 2] No such file or directory: 'favicon.ico'怎么办啊

**Reply by viopsa233** [reply]
> 大佬，出现FileNotFoundError: [Errno 2] No such file or directory: 'favicon.ico'怎么办啊

不用管

**Comment by pointm** [comment]
楼主，，你是，我的超人

**Comment by pointm** [comment]
大佬，我在手机上用的时候有时候会卡死然后收到通知：
localhost 提示：
 no close frame received or sent
Sydney的回复变成了我的输入框里面的内容，请问我该怎么办呢？

**Comment by appmethod7** [comment]
我又不会了
![Uploading Screenshot_20230513_191713_com.termux_edit_377994740247528.jpg…]()

**Reply by viopsa233** [reply]
看不见你图片啊

**Reply by appmethod7** [reply]
弹出来“ERROR : Could not open requirements file :[ Errno 
2] No such file or directory :' requirements . txt '”这个东西

**Reply by viopsa233** [reply]
> 弹出来“ERROR : Could not open requirements file :[ Errno 2] No such file or directory :' requirements . txt '”这个东西

文件缺失,看看你的解压文件夹里面有没有requirements.txt

**Reply by appmethod7** [reply]
已解决，好像是因为套娃文件夹的问题

**Comment by appmethod7** [comment]
访问的时候要不要魔法?

**Reply by viopsa233** [reply]
应该不用

**Comment by appmethod7** [comment]
这又是什么意思啊？
![Screenshot_20230513_201243_com termux](https://github.com/viopsa233/AI-testers-Forum/assets/128205751/6be2ad7c-f08c-47d7-995d-f2ebacb08b47)

**Reply by viopsa233** [reply]
HTTP请求错误,重新试试

**Reply by appmethod7** [reply]
SSL_PROTOCOL_ERROR重新试试之后弹出来这个错误，是因为什么？

**Reply by viopsa233** [reply]
> SSL_PROTOCOL_ERROR重新试试之后弹出来这个错误，是因为什么？

没遇到过,先等InterestingDarkness来先

**Comment by 20atri** [comment]
大佬，我的网页提示no close frame received or sent
请问是什么情况？![Screenshot_20230513_230247.jpg](https://github.com/viopsa233/AI-testers-Forum/assets/133430353/7fcc0511-9253-4137-8a3d-8e79b1ee006c)

**Reply by viopsa233** [reply]
> 大佬，我的网页提示no close frame received or sent
> 请问是什么情况？![Screenshot_20230513_230247.jpg](https://github.com/viopsa233/AI-testers-Forum/assets/133430353/7fcc0511-9253-4137-8a3d-8e79b1ee006c)
> 
> 

看你图片有两个报错，no close frame received or sent，应该是后台进程杀死了，第2个报错是tokens上限了

**Comment by xiyou84** [comment]
![2C2731FB89C5C9294621E27950598E6E](https://github.com/viopsa233/AI-testers-Forum/assets/122877925/9b1b2812-f7b3-4d49-a0b7-df3282ab71f4)
这是怎么个情况啊大佬

**Reply by viopsa233** [reply]
cd 都错了,是 `cd 内部存储／ChatSydney-react／ChatSydney-react`

**Comment by xiyou84** [comment]
大佬，部署一切完成了，唯一的缺点是超级慢，平均五分钟回复我一下，我该怎么办呢

**Reply by viopsa233** [reply]
Termux要挂后台

**Reply by Yimo-Senior** [reply]
python main.py
Traceback (most recent call last):
  File "/root/warehouse/ChatSydney/main.py", line 7, in <module>
    from EdgeGPT import Chatbot
  File "/root/warehouse/ChatSydney/EdgeGPT.py", line 32, in <module>
    from BingImageCreator import ImageGen
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/BingImageCreator.py", line 15, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
这个是什么问题

**Comment by lbls888** [comment]
我根据教程来做，完成后却出现ClientSession.ws](http://xn--clientsession-hn2v05onwcu47c3ryqh8e.ws/)_connect() got an unexpected keyword argument 'ping_interval'是打开聊天界面，发送字符才出现的。问了下克劳德，没有给到准确的办法。。
![Screenshot_20230518_022136](https://github.com/viopsa233/AI-testers-Forum/assets/131499083/8a48ac2c-48b8-4cfa-8f75-65a9face2d09)

**Reply by ghost** [reply]
新版EdgeGPT的问题 `pip install EdgeGPT==0.4.2`

**Reply by lbls888** [reply]
谢谢，一试就灵啊

**Reply by lbls888** [reply]
请问下，预设的提示词怎么修改？在哪个文件？

**Reply by ghost** [reply]
可以直接修改，代码在index.html

**Reply by lbls888** [reply]
代码跟搜索能力有关系吗？我觉得搜索能力好差啊，很难搜索信息的样子？

**Reply by ghost** [reply]
可能是IP的问题，需要修改EdgeGPT源码

**Reply by lbls888** [reply]
我没用VPN，照你这么说，应该是VPN问题了。谢谢。我试试

**Reply by ghost** [reply]
不是代理问题，是XFF头的问题

**Reply by lbls888** [reply]
不知道什么时候才能加强搜索能力，没了搜索能力的搜索引擎，用不习惯了。。

**Comment by appmethod7** [comment]
![Screenshot_20230521_083715_com termux](https://github.com/viopsa233/AI-testers-Forum/assets/128205751/0cbf9efc-82cb-4934-9549-82a67b3962e5)

![Screenshot_20230521_083659_com termux](https://github.com/viopsa233/AI-testers-Forum/assets/128205751/6456d770-afac-4bf8-88d7-f914480fcd80)
这啥意思啊?

**Comment by Ninokz** [comment]
同样，按照指示一步一步来，🤔termux能运行起来，但访问这边出了这种问题
![Screenshot_2023_0522_025918](https://github.com/viopsa233/AI-testers-Forum/assets/33475233/70dfd377-937b-4029-8c00-7302bcbb88b6)
[](url)

**Reply by ghost** [reply]
EdgeGPT上游变化导致旧代码不能正常工作，等大神PR修，或者装旧版EdgeGPT

**Reply by ghost** [reply]
pip install EdgeGPT==0.4.4

**Comment by falsemaskx** [comment]
大佬您好，请问显示   localhost:65432显示

'messages'   是什么情况？而且发消息总是经常被撤回，需要不停发继续，而且继续一段时间后它还会混乱，请问有什么解决办法吗？

**Reply by viopsa233** [reply]
更新一下EdgeGPT

**Reply by falsemaskx** [reply]
更新后显示localhost:65432

WebSocket error: [object Event]🤔请问大佬要怎么办？

**Reply by viopsa233** [reply]
> 更新后显示localhost:65432
> 
> WebSocket error: [object Event]🤔请问大佬要怎么办？

你的chatSydney更新了吗

**Reply by falsemaskx** [reply]
对，070

**Reply by viopsa233** [reply]
> WebSocket error: [object Event]

你把错误截图我看看

> 更新后显示localhost:65432
> 
> WebSocket error: [object Event]🤔请问大佬要怎么办？

你把错误截图我看看

**Reply by falsemaskx** [reply]
![Screenshot_2023-06-01-20-39-20-04](https://github.com/viopsa233/AI-testers-Forum/assets/122167291/d2b21199-c3eb-4e4b-a4c8-0647ac203e03)
![Screenshot_2023-06-01-19-32-32-16](https://github.com/viopsa233/AI-testers-Forum/assets/122167291/15835c39-d83d-4637-9a79-baad274204fa)

**Reply by viopsa233** [reply]
> ![Screenshot_2023-06-01-20-39-20-04](https://user-images.githubusercontent.com/122167291/242599302-d2b21199-c3eb-4e4b-a4c8-0647ac203e03.jpg) ![Screenshot_2023-06-01-19-32-32-16](https://user-images.githubusercontent.com/122167291/242599329-15835c39-d83d-4637-9a79-baad274204fa.jpg)

挂梯子,EDGEgpt版本回退试试`pip install EdgeGPT==0.6.5`

**Reply by falsemaskx** [reply]
请问手机版是无法使用070版本吗？只能够使用旧版？

**Reply by ghost** [reply]
你的ChatSydney不是最新的，最新推荐EdgeGPT版本是0.6.5

**Reply by ghost** [reply]
这个ChatSydney从UI看就很老

**Reply by falsemaskx** [reply]
但是如果手机安装065，也会出现上面的问题

**Comment by NocturnalRushers** [comment]
大佬，出现这种情况怎么办
![Uploading -4f8b20091c7e8341.png…]()

**Reply by NocturnalRushers** [reply]
在http://localhost:65432页面聊天框输入文字后，页面弹窗显示: string indices must be intergers, not 'str'

**Reply by viopsa233** [reply]
> 在http://localhost:65432页面聊天框输入文字后，页面弹窗显示: string indices must be intergers, not 'str'

没见过这个报错,重新按步骤安装试一下

**Reply by NocturnalRushers** [reply]
> > 在[http://localhost:65432页面聊天框输入文字后，页面弹窗显示](http://localhost:65432%E9%A1%B5%E9%9D%A2%E8%81%8A%E5%A4%A9%E6%A1%86%E8%BE%93%E5%85%A5%E6%96%87%E5%AD%97%E5%90%8E%EF%BC%8C%E9%A1%B5%E9%9D%A2%E5%BC%B9%E7%AA%97%E6%98%BE%E7%A4%BA): string indices must be intergers, not 'str'
> 
> 没见过这个报错，重新按步骤安装试一下

试了一下，重装无用。听别人说可能是cookies.json文件有误，我将文件删了，直接输入_U cookie，然后成功了。

**Comment by KeLevery** [comment]


**Reply by viopsa233** [reply]
pip和Python是不是最新的

**Comment by KeLevery** [comment]


**Reply by KeLevery** [reply]
Task exception was never retrieved
future: <Task finished name='Task-25' coro=<RequestHandler.start() done, defined at /data/data/com.termux/files/usr/lib/python3.11/site-packages/aiohttp/web_protocol.py:462> exception=AssertionError()>
Traceback (most recent call last):
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 505, in start
    request = self._request_factory(message, payload, self, writer, handler)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/aiohttp/web_app.py", line 446, in _make_request
    return _cls(
           ^^^^^
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/aiohttp/web_request.py", line 811, in __init__
    super().__init__(*args, **kwargs)
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/aiohttp/web_request.py", line 188, in __init__
    assert transport is not None
           ^^^^^^^^^^^^^^^^^^^^^
AssertionError

**Reply by KeLevery** [reply]
感谢🙏作者大大问题已经解决了

**Comment by Fgxc1** [comment]
请问为什么经常会出现"Throttled: Request is throttled" 换梯子也没用 只有换cookies才能继续聊

**Reply by ghost** [reply]
已经加入300条用量上限突破，现在上限未知

**Comment by Fgxc1** [comment]
好的



---Original---
From: ***@***.***&gt;
Date: Mon, Jun 5, 2023 09:41 AM
To: ***@***.***&gt;;
Cc: "Dayang ***@***.******@***.***&gt;;
Subject: Re: [viopsa233/AI-testers-Forum] ChatSydney To Termux /OriginalAuthor By InterestingDarkness (Discussion #44)




 
已经加入300条用量上限突破，现在上限未知

**Comment by overKAC** [comment]
大佬这是什么情况呜呜呜
$ python main.py
Traceback (most recent call last):
  File "/storage/emulated/0/ChatSydney/main.py", line 78, in <module>
    loaded_cookies = json.load(f)
                     ^^^^^^^^^^^^
  File "/data/data/com.termux/files/usr/lib/python3.11/json/__init__.py", line 293, in load
    return loads(fp.read(),
           ^^^^^^^^^^^^^^^^
  File "/data/data/com.termux/files/usr/lib/python3.11/json/__init__.py", line 335, in loads
    raise JSONDecodeError("Unexpected UTF-8 BOM (decode using utf-8-sig)",
json.decoder.JSONDecodeError: Unexpected UTF-8 BOM (decode using utf-8-sig): line 1 column 1 (char 0)

**Reply by ghost** [reply]
cookies.json太麻烦就别用了，网页上也能输入_U

**Reply by overKAC** [reply]
呜呜大佬能不能教一下怎么搞

**Comment by Yimo-Senior** [comment]


**Reply by Yimo-Senior** [reply]
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130832815/29551fad-d691-411e-a3cf-cabb08179e44)

**Reply by Yimo-Senior** [reply]
解决了

**Comment by GarrinMartz5992** [comment]
![Screenshot_2023-06-15-10-02-38-877_com termux](https://github.com/viopsa233/AI-testers-Forum/assets/116739059/2e1c9f84-1fba-4d7a-a85c-cbcdd1061e19)
大佬这咋办

**Reply by Yimo-Senior** [reply]
按这个步骤来试试。我是这样安装成功的
必备魔法上网
1.apt update && apt upgrade
2.pkg install git
3.pkg install python-pip
4.pip install setuptools
5.termux-setup-storage
6.git clone https://github.com/InterestingDarkness/ChatSydney.git
7.cd ChatSydney
8.pip install -r requirements.txt
9.python main.py
得到一个链接：http://localhost:65432
10.去浏览器输入这个链接打开，如果浏览器一直白屏，去下载edge试试。我的本地浏览器和qq浏览器都打不开，换了edge打开的。
11.然后在电脑上获取cookies，不建议使用cookies.json，太麻烦了。直接使用另一种方法，获取这个u里面的值。然后放到网页中的哪个_U cookie中。就可以聊天了。
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130832815/2fd85507-6767-4ee5-bd3f-9ab379996db9)

按照这个步骤来，不用操作其他步骤，直接复制粘贴就行，就最后一步需要你获取cookie
有电脑的话，还是在电脑上用，手机上token多了之后，5分钟一条消息，还容易人工智障，一直回复一个问题，无论你问什么。

**Reply by Yimo-Senior** [reply]
我安装这个的时候，软链接好像并不是必要的，可以不用软链接，直接在termux中拉取仓库项目，然后安装依赖，就可以了

**Reply by Yimo-Senior** [reply]
问题出现最多的就是cookies上，使用cookies.json出现的问题最多，使用暗黑哥也推荐直接使用ucookie

**Comment by viopsa233** [comment]
> 我安装这个的时候，软链接好像并不是必要的，可以不用软链接，直接在termux中拉取仓库项目，然后安装依赖，就可以了




之前没有直接填入U_这个功能需要,现在需要不需要都行

**Comment by CICSilver** [comment]
黑暗大佬删号了？怎么chatsydney的库都找不到了

**Reply by viopsa233** [reply]
是的

**Comment by falsemaskx** [comment]
![Screenshot_2023-06-28-20-18-36-00](https://github.com/viopsa233/AI-testers-Forum/assets/122167291/fec382e2-4490-4900-9a89-378ee21d3f6e)
请问大佬这个是什么情况

**Reply by viopsa233** [reply]
去官网人机验证

**Reply by xuyufenfei** [reply]
我也这样，之前那个账号死活过不了验证，换了个账号，没有报验证，但还是用不了

**Reply by falsemaskx** [reply]
请问大佬，如何在官网人机验证？没有发现

**Reply by viopsa233** [reply]
或者你可以试试别人搭建好的

**Comment by viopsa233** [comment]
> 请问大佬，如何在官网人机验证？没有发现

用网页的,不是应用的

**Reply by falsemaskx** [reply]
的确用的是的电脑网页的bing

**Reply by falsemaskx** [reply]
但是没有看到哪里验证

**Reply by viopsa233** [reply]
> 但是没有看到哪里验证

你发一句话看看

**Reply by xuyufenfei** [reply]
我的有验证，但一直失败

**Reply by viopsa233** [reply]
> 我的有验证，但一直失败

梯子问题？

**Reply by xuyufenfei** [reply]
换了个账号好了，然后我用这个账号的cookie去登，结果又提示验证了，等放假在电脑上试试

**Comment by xuyufenfei** [comment]
![Screenshot_2023-07-14-02-20-29-067_com lemurbrowser exts](https://github.com/viopsa233/AI-testers-Forum/assets/133896005/2597343b-cd38-48ba-901a-b29948740a9f)

**Comment by qyh9527** [comment]
![mmexport1690356332497](https://github.com/viopsa233/AI-testers-Forum/assets/134232728/52ae6480-2191-4747-b664-0f249f317579)

**Reply by qyh9527** [reply]
这样该如何解决？

**Comment by qyh9527** [comment]
> ![mmexport1690356332497](https://user-images.githubusercontent.com/134232728/256150575-52ae6480-2191-4747-b664-0f249f317579.jpg)
